extend-expect.d.ts

```diff
@@ -1,7 +1,9 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```

Comment:
Why don't we have `react-test-renderer` in our `dependencies`? 🤔
We're obviously using it.

If we want to have it in our `peerDependencies`, we should also have it in our `devDependencies`

Reply:
We want to use user's RTR and not potentially import another version. I've added React, RN & RTR to peer deps with the same limits as in RNTL.